event/consumer_test.go

```diff
@@ -2,138 +2,157 @@ package event_test
-					<-validMessage.UpstreamDone()
-					<-invalidMessage.UpstreamDone()
-					So(len(validMessage.CommitCalls()), ShouldEqual, 1)
-					So(len(invalidMessage.CommitCalls()), ShouldEqual, 1)
-					So(len(validMessage.ReleaseCalls()), ShouldEqual, 1)
-					So(len(invalidMessage.ReleaseCalls()), ShouldEqual, 1)
-				})
-			})
-		})
+			message1 := kafkatest.NewMessage([]byte(marshal(expectedEvent1)), 0)
+			messageConsumer.Channels().Upstream <- message1
+
+			message2 := kafkatest.NewMessage([]byte(marshal(expectedEvent)), 0)
+			messageConsumer.Channels().Upstream <- message2
+
+			message3 := kafkatest.NewMessage([]byte(marshal(expectedEvent)), 0)
+			messageConsumer.Channels().Upstream <- message3
+
+			message4 := kafkatest.NewMessage([]byte(marshal(expectedEvent)), 0)
+			messageConsumer.Channels().Upstream <- message4
 
-		Convey("With a failing handler and a kafka message with the valid schema being sent to the Upstream channel", func() {
-			mockEventHandler.HandleFunc = func(ctx context.Context, config *config.Config, event *event.HelloCalled) error {
-				defer handlerWg.Done()
-				return errHandler
-			}
-			message := kafkatest.NewMessage(marshal(testEvent), 0)
-			mockConsumer.Channels().Upstream <- message
-
-			Convey("When consume message is called", func() {
-
-				handlerWg.Add(1)
-				event.Consume(testCtx, mockConsumer, mockEventHandler, &config.Config{KafkaNumWorkers: 1})
-				handlerWg.Wait()
-
-				Convey("An event is sent to the mockEventHandler ", func() {
-					So(len(mockEventHandler.HandleCalls()), ShouldEqual, 1)
-					So(*mockEventHandler.HandleCalls()[0].HelloCalled, ShouldResemble, testEvent)
-				})
-
-				Convey("The message is committed and the consumer is released", func() {
-					<-message.UpstreamDone()
-					So(len(message.CommitCalls()), ShouldEqual, 1)
-					So(len(message.ReleaseCalls()), ShouldEqual, 1)
-				})
+			<-eventHandler.EventUpdated
+
+			Convey("The expected event is sent to the handler", func() {
+				So(len(eventHandler.Events), ShouldEqual, 4)
+
+				actual := eventHandler.Events[0]
+				So(actual.DataType, ShouldEqual, expectedEvent1.DataType)
+				So(actual.Title, ShouldEqual, expectedEvent1.Title)
+			})
+			Convey("The message is committed and the consumer is released", func() {
+				<-message1.UpstreamDone()
+				So(len(message4.CommitCalls()), ShouldEqual, 1)
+				So(len(message4.ReleaseCalls()), ShouldEqual, 1)
 			})
 		})
+		consumer.Close(nil)
 	})
 }
 
-// marshal helper method to marshal a event into a []byte
-func marshal(event event.HelloCalled) []byte {
-	bytes, err := schema.HelloCalledEvent.Marshal(event)
-	So(err, ShouldBeNil)
-	return bytes
+func TestClose(t *testing.T) {
+
+	Convey("Given a consumer", t, func() {
+		messageConsumer := kafkatest.NewMessageConsumer(false)
+		eventHandler := eventtest.NewEventHandler()
+		cfg, err := config.Get()
+		if err != nil {
+			t.Log(ctx, "failed to retrieve configuration", err)
+			t.Fail()
```

Comment:
Again use t.Fatalf() instead of t.Log and t.Fail

Reply:
Please check for other instance of this

Reply:
Think about where cfg gets instantiated in the tests as you should be able to reuse the same variable, e.g. before first Convey(...) - you wont want to do this if you are manipulating the default values..... I would suggest you do manipulated the batch size default to what is needed in the test, otherwise you are dependent on the configuration defaults set in config.go or those pulled in by the environment, which could be different to what is expected for test.